File: C++_program/Problem_on_Numbers/AutomorphicNumber.cpp
//////////////////////////////////////////////////////////////////////////////////////////////
//
//required header file
//
//////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>                   // for taking input from user
using namespace std ;

//////////////////////////////////////////////////////////////////////////////////////////////
//
//  Class Name  :   Logic
//  Function Name : AutomorphicNumber()
//  Description :   check number is Automorphic number or not
//  Input :         Integer
//  output :        void
//  Author :        Rekha Shankarlal Kumawat
//  Date :          22/03/2025
//
//////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std ;

class Logic
{

    public:
        int iNo;

        Logic(int iNo)
        {
            this->iNo = iNo;
        }
        bool AutomorphicNumber()
        {
            int iCount = 0 ;
            int iTemp = 0 ;
            int iSqr = 0 ;
            int iDeno = 0;
            int iCnt = 0 ;

            iTemp = this->iNo ;

            while (this->iNo != 0) 
            {
                iCount++ ;
                this->iNo = this->iNo / 10;
                
            }

            this->iNo = iTemp;
            iSqr = this->iNo * this->iNo;
            iDeno = 1 ;
            for(iCnt = 0; iCnt < iCount; iCnt++)
            {
                iDeno = iDeno * 10;
            }

            return((iSqr % iDeno) == iTemp);  
        }
};
int main()
{
    int iValue = 0;
    bool bRet = false ;

    cout<<"Enter the Number :- ";
    cin>>iValue;
    
    Logic lobj(iValue);
    bRet =lobj.AutomorphicNumber();

    if(bRet == true)
    {
        cout<<iValue<<" is a Automorphic Number\n";
    }
    else
    {
        cout<<iValue<<" is not a Automorphic Number\n";
        
    }
    return 0 ;
}

File: Python_Programming/Problems_on_Numbers/AutomorphicNumber.py
'''
    Description : program  to check the number is Automorphic number or not 
                EX :- 25 -> 25^2 = 625  (last digit of ans is 25 so its Automorphic)
                
    Date :- 21/03/2026
    
    Author : Rekha Shankarlal Kumawat
                
'''

def AutomorphicNumber(iNo):
    
    iCount = 0 
    iTemp = 0 
    iSqr = 0 
    iDeno = 0
    iCnt = 0 

    iTemp = iNo 

    while (iNo != 0):
    
        iCount = iCount +1 
        iNo = iNo // 10
        
    

    iNo = iTemp
    iSqr = iNo * iNo
    iDeno = 1 
    for i in range(1 ,iCount+1):
        iDeno = iDeno * 10

    return((iSqr % iDeno) == iTemp)
    

def main():
    
    iValue = 0
    bRet = False 

    print("Enter the Number :- ")
    iValue = int(input())
    
    bRet =  AutomorphicNumber(iValue)

    if(bRet == True):
        print(iValue ,"is a Automorphic Number")
    else:
        print(iValue ,"is not a Automorphic Number")

if __name__ == "__main__":
    main()